test/inference/hmc.jl

```diff
@@ -173,4 +174,12 @@ include(dir*"/test/test_utils/AllUtils.jl")
         @test size(c1, 1) == 400
         @test size(c2, 1) == 500
     end
+    @turing_testset "AHMC resize" begin
```

Comment:
Does this bug only occurs when HMC is used in Gibss?

Reply:
Yes, but I'm not sure why. Gibbs calls `sample_init!` on all the samplers so the `update` line is called for sure. And why only NUTS? Why not HMC/HMCDA?

Reply:
> Yes, but I'm not sure why. Gibbs calls sample_init! on all the samplers so the update line is called for sure. 

Not sure about this one. I'm looking into it.

> And why only NUTS? Why not HMC/HMCDA?

NUTS by default is using `DiagEuclideanMetric` but the other two are using `UnitEuclideanMetric` which is dimension agnoistic.